astropy/nddata/blocks.py

```diff
@@ -0,0 +1,208 @@
+
+    block_size : int or array_like (int)
+        The integer block size along each axis.  If ``block_size`` is a
+        scalar and ``data`` has more than one dimension, then
+        ``block_size`` will be used for for every axis.  Each dimension
+        of ``block_size`` must divide evenly into the corresponding
+        dimension of ``data``.
+
+    Returns
+    -------
+    output : `~numpy.ndarray`
+        The reshaped array as a view of the input ``data`` array.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from astropy.nddata import reshape_as_blocks
+    >>> data = np.arange(16).reshape(4, 4)
+    >>> data
+    array([[ 0,  1,  2,  3],
+           [ 4,  5,  6,  7],
+           [ 8,  9, 10, 11],
+           [12, 13, 14, 15]])
+    >>> reshape_as_blocks(data, (2, 2))
+    array([[[[ 0,  1],
+             [ 4,  5]],
+            [[ 2,  3],
+             [ 6,  7]]],
+           [[[ 8,  9],
+             [12, 13]],
+            [[10, 11],
+             [14, 15]]]])
+    """
+
+    data, block_size = _process_block_inputs(data, block_size)
+
+    if np.any(np.mod(data.shape, block_size) != 0):
+        raise ValueError('Each dimension of block_size must divide evenly '
+                         'into the corresponding dimension of data')
+
+    nblocks = np.array(data.shape) // block_size
+    new_shape = tuple(k for ij in zip(nblocks, block_size) for k in ij)
+    nblocks_idx = tuple(range(0, len(new_shape), 2))  # even indices
+    block_idx = tuple(range(1, len(new_shape), 2))  # odd indices
+
+    return data.reshape(new_shape).transpose(nblocks_idx + block_idx)
+
+
+@support_nddata
+def block_reduce(data, block_size, func=np.sum):
+    """
+    Downsample a data array by applying a function to local blocks.
+
+    If ``data`` is not perfectly divisible by ``block_size`` along a
+    given axis then the data will be trimmed (from the end) along that
+    axis.
+
+    Parameters
+    ----------
+    data : array_like
+        The data to be resampled.
+
+    block_size : int or array_like (int)
+        The integer block size along each axis.  If ``block_size`` is a
+        scalar and ``data`` has more than one dimension, then
+        ``block_size`` will be used for for every axis.
+
+    func : callable, optional
+        The method to use to downsample the data.  Must be a callable
+        that takes in a `~numpy.ndarray` along with an ``axis`` keyword,
+        which defines the axis or axes along which the function is
+        applied.  The ``axis`` keyword must accept multiple axes as a
+        tuple.  The default is `~numpy.sum`, which provides block
+        summation (and conserves the data sum).
+
+    Returns
+    -------
+    output : array_like
+        The resampled data.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from astropy.nddata import block_reduce
+    >>> data = np.arange(16).reshape(4, 4)
+    >>> block_reduce(data, 2)    # doctest: +SKIP
```

Comment:
Wait... why is it still skipped but my comment marked as resolved? Hope you can clarify. Thanks.

Reply:
Hmm, I accepted your suggestion above.

Reply:
I just fixed that (again).